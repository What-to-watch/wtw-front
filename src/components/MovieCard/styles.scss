.movie-card {
    width: 166px;
    height: 239px;
    position: relative;
    overflow: hidden;
    transition: all ease-out 150ms;
    &__background {
        width: 100%;
        height: 100%;
        position: absolute;
        &__gradient {
            transition: all ease-out 150ms;
            position: absolute;
            width: 100%;
            height: 100%;
            opacity: 1;
        }
        &__image{
            position: absolute;
            width: 100%;
            height: 100%;
            background: url('/movie-posters/LoadPoster.png') no-repeat scroll 0 0;
            background-size: cover;
        }
    }
    &__info{
        position: absolute;
        transition: all ease-out 150ms;
        opacity: 1;
        width: 166px;
        height: 239px;
        display: flex;
        flex-direction: column;
        justify-content: flex-end;
        & h2, & p {
            color: white;
            padding: 0 8px;
        }
        & h2 {
            margin: 0;
            font-size: 24px;
            font-weight: 900;
            line-height: 1.1em;
        }
        & p {
            margin-top: 0;
            font-size: 14px;
            width: 166px;
            overflow: hidden;
            white-space: nowrap;
            text-overflow: ellipsis;
            margin-bottom: 16px;
        }
    }
}

.movie-card:hover {
    box-shadow: -4px 4px 32px rgba(0, 0, 0, 0.32), -3px 3px 24px rgba(0, 0, 0, 0.4), -2px 2px 16px rgba(0, 0, 0, 0.48);
    & .movie-card__background {
        &__gradient {
            opacity: 0;
        }
    }
    & .movie-card__info { 
        opacity: 0;
    }
}