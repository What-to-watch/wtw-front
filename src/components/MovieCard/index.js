import React from 'react';
import PropTypes from 'prop-types';
import getRandomHue from '../../utils/getRandomHue';

import './styles.scss';

const getGradient = (hue) => {
    return `linear-gradient(180deg, hsla(${hue}, 100%, 50%, 0) 0%, hsla(${hue}, 100%, 10%, 0.8) 56%, hsla(${hue}, 100%, 0%, 0.9) 100%)`;
}

const MovieCard = ({...props}) => {
    const { title, genres, id, src, hue = getRandomHue(), onClick = x=>x } = props;
    const genresString = genres.join('/');
    const gradient = getGradient(hue);
    const handleClick = () => {
        onClick(id);
    }
    return(
        <a className="movie-card" href={`#${id}`} onClick={handleClick}>
            <div className="movie-card__background" >
                <img 
                    className="movie-card__background__image" 
                    src={src ? src : '/movie-posters/NoPoster.png'}
                    alt={title}
                />
                <div className={"movie-card__background__gradient "}  style={{backgroundImage:gradient}}></div>
                
            </div>
            <div className="movie-card__info">
                <h2>{title}</h2>
                <p>{genresString}</p>
            </div>
        </a>
    );
}

export const LoadingMovieCard = (props) => {
    const { hue = getRandomHue() } = props;
    const gradient = getGradient(hue);
    return (
        <div className="loading-movie-card">
            <div className="loading-movie-card__dots">
            .
            </div>
            <div className={"loading-movie-card__gradient"}  style={{backgroundImage:gradient}}></div>
        </div>
    )
}

MovieCard.defaultProps = {
    title: 'Movie1',
    genres: ['Comedy'],
    id: 1,
    src: '/movie-posters/NoPoster.png',
};

MovieCard.propTypes = {
    title: PropTypes.string,
    genres: PropTypes.array,
    id:PropTypes.number,
    src: PropTypes.string,
    hue: PropTypes.number,
};

export default MovieCard;