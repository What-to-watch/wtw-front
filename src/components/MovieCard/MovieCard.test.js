import React from 'react';
import MovieCard from './index';
import renderer from 'react-test-renderer';

it('MovieCard render correctly', ()=>{
    const props = {
        title: "movie1",
        genres: ['Action', 'Comedy'],
        hue: 200,
        id: 1
    }
    const component = renderer.create(
        (<MovieCard {...props}/>),
    );

    let moviecard = component.toJSON();
    const titleNode = component.root.findByType("h2");
    const genresNode = component.root.findByType("p");

    expect(titleNode.children[0]).toBe(props.title);

    expect(genresNode.children[0]).toBe(props.genres.join('/'));

    expect(moviecard).toMatchSnapshot();
})