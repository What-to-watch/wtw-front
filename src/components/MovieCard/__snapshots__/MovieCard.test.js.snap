// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<a
  className="movie-card"
  href="#1"
>
  <div
    className="movie-card__background"
  >
    <div
      className="movie-card__background__gradient "
      style={
        Object {
          "backgroundImage": "linear-gradient(180deg, hsla(200, 100%, 50%, 0) 0%, hsla(200, 100%, 10%, 0.8) 56%, hsla(200, 100%, 0%, 0.9) 100%)",
        }
      }
    />
    <img
      alt="movie1"
      className="movie-card__background__image"
      src="/movie-posters/NoPoster.png"
    />
  </div>
  <div
    className="movie-card__info"
  >
    <h2>
      movie1
    </h2>
    <p>
      Action/Comedy
    </p>
  </div>
</a>
`;

exports[`MovieCard render correctly Movie Card default value 1`] = `
<a
  className="movie-card"
  href="#1"
>
  <div
    className="movie-card__background"
  >
    <div
      className="movie-card__background__gradient "
      style={
        Object {
          "backgroundImage": "linear-gradient(180deg, hsla(242, 100%, 50%, 0) 0%, hsla(242, 100%, 10%, 0.8) 56%, hsla(242, 100%, 0%, 0.9) 100%)",
        }
      }
    />
    <img
      alt="Movie1"
      className="movie-card__background__image"
      src="/movie-posters/NoPoster.png"
    />
  </div>
  <div
    className="movie-card__info"
  >
    <h2>
      Movie1
    </h2>
    <p>
      Comedy
    </p>
  </div>
</a>
`;
